### DISCUSSION TITLE
FR: Entrification: Ability to create "Groups" for Entry Types

### DISCUSSION THREAD

**Discussion by adrianjean** [craftcms/cms#13307] [Ideas] [answered]
I have been embracing the [Entrification](https://craftcms.com/blog/entrification) of **Categories**, **Tags**, and **Global Sets** and really think it all makes sense. The issue I have run into is the organization of all these new Entry Types in the CP. 

One of the conveniences of Global Sets being in a different area of the website, was that it was logical for clients to go to a "Globals" section to make changes that had far reaching effects. Same thing with Categories.

Now, having converted non-entry types into Entries, the Entry Type listing is getting quite long (See Screenshot).

Ideally we would have some way to organize these types in the sidebar to make things simpler to navigate. Like "Groups" or "Folders" or collapsible hierarchical structures that allow us to group & hide Entry Types to clean up the list a bit.

Thoughts?

<img width="218" alt="Screenshot 2023-06-12 at 6 44 07 PM" src="https://github.com/craftcms/cms/assets/3161182/eafad2a7-4943-4f29-9f03-4a31b182d2ef">

**Comment by brandonkelly** [comment]
We’re exploring ways to improve this for Craft 5 ;)

**Comment by davidhellmann** [comment]
Just a short one... An improvement would be to make the headings collapsable by default. Maybe a light switch for Open/Closed by default. Then, it's saved to the user. I've added a bit of control-panel CSS to reduce the size of the items. But the possibility of folding the blocks would be a big update. Like the new sidebar :) A border-bottom to the headlines would also be great for more optical separation of the sections. 

![image](https://github.com/craftcms/cms/assets/764292/e79112bd-22d3-46dd-9dbc-0d84896e653b)

**Reply by davidhellmann** [reply]
![CleanShot 2024-08-22 at 10 09 13](https://github.com/user-attachments/assets/7996ed70-a4e0-43e7-88f3-dd3374711a1a)

I also added an open/close to the groups. I checked the site parameter on the URL and saved this stuff in local storage.

```js
openHeadings: {"marketing_en":[0],"marketing_de":[0,3,2],"careers_en":[0,2]}
```

Feedback is good!

**Reply by brandonkelly** [reply]
Craft 5.8.0 is out with collapsible source headings (#17226).

**Comment by mmikkel** [comment]
Related: https://github.com/craftcms/cms/discussions/14179

**Reply by richhayler** [reply]
@brandonkelly as you are exploring this for Craft 5, please could you consider a way to have an entry type be a member of multiple groups.

As entry types can now be used across more than one section having the flexibility to organise them by which section they are associated with for example, would be useful.

**Comment by russback** [comment]
Related perhaps... 

Where we many entry types (30+) for a site that has lots of page templates and content components using Matrix, we're finding it can get quite confusing from a DX point of view, as there's one flat list of entry types. Icons are kind of helpful but we're having to use naming conventions to reduce the cognitive load in trying to figure out the different kind of entry types.

One suggestion from the team was to be able to organise entry types into folders, or categories. Basically some kind of way to more clearly organise entry types when building out the site.

An example might be a Slideshow component that gets used on various pages. That's an entry type with fields like title, design theme, etc, and then a field for each slide, which is a matrix of Slide entry types. So we might have something like this when viewing available entry types:

Pages
- Home
- Article
- Blog
- Landing
Components
- Slideshow
- Video
- Contact Form
Atoms
- Slides

Ideally the names of categories/folders would be customisable to fit with a preferred approach for a design/content system.

**Comment by brandonkelly** [comment, marked as answer]
Craft 5.9.0 is out with the ability to create custom entry index pages (#17779) 🎉
